### ISSUE TITLE
Build failure: pcsx2

### ISSUE THREAD

**Issue by matj1** [NixOS/nixpkgs#287704]
### Steps To Reproduce

1. Build pcsx2.

Specifically, I have
```nix
  users = {
    users.matj1 = {
      packages = with pkgs; [
        …
        pcsx2
        …
      ];
    };
  };
```
in `/etc/nixos/configure.nix` and run `sudo nixos-rebuild switch`.

### Build log

```
…
error: hash mismatch in fixed-output derivation '/nix/store/zra02v4kwfkb87rd5fgi0wjssg20z1q9-source.drv':
         specified: sha256-5ZCXw6PEQ6Ed6kEP27m9O0U79uVGEFR/vwee6/dZBD8=
            got:    sha256-9Zorm+rTXZwiBeFf0OfhAJuP4ebT/HnIx9F6H+pxj8M=
error: 1 dependencies of derivation '/nix/store/jcxfnh8clz6dz37ffr2nljwra6zqkbwd-pcsx2-1.7.5474.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mn8jkg76z3cjfhjd5fxfks7a7y82f0j4-user-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3d47zhnc0ya0nh1knvgr281zd8sijcz2-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/j00xibqvpwfn3mdkpznhb9xscknnwmja-nixos-system-nixos-24.05pre576949.c002c6aa977a.drv' failed to build
```

The full log is [here](https://gist.github.com/matj1/082aec73313b0475af515c0e7cbe59c4).

### Notify maintainers

@hrdinka  
@GovanifY

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

- system: `"x86_64-linux"`
- host os: `Linux 6.1.74, NixOS, 24.05 (Uakari), 24.05pre576949.c002c6aa977a`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.18.1`
- channels(root): `"nixos"`
- nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by SuperSamus** [comment]
This should have been fixed already: #282953.
Is your nixpkgs updated?

**Comment by matj1** [comment]
Indeed, it isn't. I thought that `nixos-rebuild switch` also automatically updates packages, but it seems that `--upgrade` is required for that. I ran that with `--upgrade`, and it works now.
